improvement(Stock Mouvement Document)

lomamech opened the pull request:
Set Voucher reference to
- Adjustment Receipt
- Donation Receipt
- Integration Receipt
- Purchase Receipt
- Loss Receipt
- Patient Receipt

closes #2694

lomamech committed:
improvement(Stock Mouvement Document)
Set Voucher reference to
- Adjustment Receipt
- Donation Receipt
- Integration Receipt
- Purchase Receipt
- Loss Receipt
- Patient Receipt

closes #2694

jniles commented on server/controllers/stock/reports/stock/adjustment_receipt.js:
```diff
@@ -28,7 +28,23 @@ async function stockAdjustmentReceipt(documentUuid, session, options) {
   `;
 
-  const rows = await db.exec(sql, [db.bid(documentUuid)]);
+  const sqlGetVoucherReference = `
```
Does every receipt need `sqlGetVoucherReference`?  If so, I would make it a function and put it in the `stock/reports/common.js`.

Like this:
```js
function getVoucherReferenceForStockMovement(documentUuid) {
  const sql = `
    SELECT v.uuid, dm.text AS voucher_reference
    FROM voucher AS v
    JOIN voucher_item AS vi ON vi.voucher_uuid = v.uuid
    JOIN document_map AS dm ON dm.uuid = v.uuid
    WHERE vi.document_uuid = ?
    LIMIT 1;
  `;

  return db.exec(sql, [db.bid(documentUuid)]);
}

exports.getVoucherReferenceForStockMovement = getVoucherReferenceForStockMovement;
```

Then every stock module can import it and just use:
```js
const { getVoucherReferenceForStockMovement }  = require('../common');

// later ..

const results = await q.all([
  db.exec(sql, [db.bid(documentUuid)]), 
  getVoucherReferenceForStockMovement(documentUuid),
]);

// etc
```

This would be more [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).

jniles commented on server/controllers/stock/reports/stock/entry_purchase_receipt.js:
```diff
@@ -38,8 +38,23 @@ function stockEntryPurchaseReceipt(documentUuid, session, options) {
+    LIMIT 1;
+  `;
+
+  return q.all([
```
By the way, you don't need to use `q` here.  You can just call the native `Promise.all()` directly.

jniles reviewed (commented):
Made a few comments on the code - haven't checked the functionality yet, but the idea is the right one!

lomamech committed:
improvement(Stock Movement Link Voucher Transaction)
- Set the function getVoucherReferenceForStockMovement in
  stock/reports/common.js

lomamech committed:
Clear and Sanitaze

jniles reviewed (approved):
Looks good to me.  Thanks a ton!
